catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitor.java

```diff
@@ -74,32 +96,54 @@ public ContentDirectoryMonitor(CamelContext camelContext) {
      * also called whenever an existing route is updated.
      */
     public void init() {
-        LOGGER.trace("INSIDE: init()");
-
         if (routeCollection != null) {
             try {
                 // This stops the route before trying to remove it
                 LOGGER.debug("Removing {} routes", routeCollection.size());
                 camelContext.removeRouteDefinitions(routeCollection);
             } catch (Exception e) {
-                LOGGER.info("Unable to remove Camel routes from Content Directory Monitor", e);
+                LOGGER.warn("Unable to remove Camel routes from Content Directory Monitor", e);
             }
         } else {
             LOGGER.debug("No routes to remove before configuring a new route");
         }
 
-        org.codice.ddf.security.common.Security.runAsAdmin(() -> {
-            configureCamelRoute();
+        runAsAdmin(() -> {
+            if (StringUtils.isEmpty(monitoredDirectory)) {
+                LOGGER.warn("Cannot setup camel route - must specify a directory to be monitored");
+                return null;
+            }
+            if (configureAsynchronously()) {
```

Comment:
That might get messy. I think `runAsync` always returns immediately regardless of the executor we use. Tests would fail unless I added synchronization mechanisms everywhere to wait for the serial executor to finish.